src/main/java/net/datafaker/providers/base/Aviation.java

```diff
@@ -110,7 +110,7 @@ public String engineType() {
      */
     public String flight(String type) {
         String airline;
-        if (type.equals("ICAO")) {
+        if ("ICAO".equals(type)) {
```

Comment:
Maybe a non-null annotation would be better?

Reply:
for this one probably both options are ok

at the same time there are still a number of other public methods taking args which potentially could be null

and there we might think about annotations or something like that... 
however the main thing is to make them tested and returning a human friendly message in case of problem